### PR TITLE
 CCD-3728: Setting up Redis infrastructure

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
CCD-3728: Setting up Redis infrastructure
          [CCD-3728] (https://tools.hmcts.net/jira/browse/CCD-3728).

### CODE REVIEW

File: redis.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+resource "azurerm_key_vault_secret" "redis_connection_string" {
+  name         = "ccd-redis-password"
+  value        = module.redis-activity-service.access_key
+  key_vault_id = data.azurerm_key_vault.shared.id
+}
+
+module "redis-activity-service" {
+  source      = "git@github.com:hmcts/cnp-module-redis?ref=master"
+  product     = "${var.product}-activity-service"
```
Comment (danlysiak) [inline redis.tf:20]:
```suggestion
module "redis-ccd" {
  source      = "git@github.com:hmcts/cnp-module-redis?ref=master"
  product     = "${var.product}-redis"
```

File: redis.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+data "azurerm_key_vault" "shared" {
+  name                = "ccd-${var.env}"
+  resource_group_name = "ccd-shared-${var.env}"
+}
+
+resource "azurerm_key_vault_secret" "redis_connection_string" {
+  name         = "ccd-redis-password"
+  value        = module.redis-activity-service.access_key
```
Comment (danlysiak) [inline redis.tf:14]:
```suggestion
  value        = module.redis-ccd.access_key
```

File: redis.tf (hunk 1 above)
Comment (Thor-tech-of-metal) [reply redis.tf:20]: DONE

File: redis.tf (hunk 2 above)
Comment (Thor-tech-of-metal) [reply redis.tf:14]: DONE

**Review by danlysiak — APPROVED** [review]

**Pull request merged**